src/control/common/test/libdaos_mocks/libdaos_mocks.h

```diff
@@ -0,0 +1,123 @@
+/* Code generated by cmd/cgo; DO NOT EDIT. */
+
+/* package github.com/daos-stack/daos/src/control/common/test/libdaos_mocks */
+
+
+#line 1 "cgo-builtin-export-prolog"
+
+#include <stddef.h> /* for ptrdiff_t below */
+
+#ifndef GO_CGO_EXPORT_PROLOGUE_H
+#define GO_CGO_EXPORT_PROLOGUE_H
+
+#ifndef GO_CGO_GOSTRING_TYPEDEF
+typedef struct { const char *p; ptrdiff_t n; } _GoString_;
+#endif
+
+#endif
+
+/* Start of preamble from import "C" comments.  */
+
+
+#line 20 "libdaos_mocks.go"
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+
+
+
+#include <daos.h>
+
+#include "daos_hdlr.h"
+
+// typedefs to work around cgo's limitations
+typedef const char const_char;
+typedef char const *const * char_const_in_arr;
```

Comment:
(style)  do not add new typedefs
(style)  "foo *const * bar" should be "foo *const *bar"